src/graph.rs

```diff
@@ -4848,110 +4848,141 @@ impl<'a> NpmSpecifierResolver<'a> {
                 NpmLoadError::NotSupportedEnvironment.into(),
               )),
             );
           }
         }
       }
+
+      if items_by_req.is_empty() {
+        break; // nothing to do, so exit
+      }
+
+      if let Some(npm_resolver) = &self.npm_resolver {
+        let all_package_reqs = items_by_req.keys().copied().collect::<Vec<_>>();
+        let resolutions =
+          npm_resolver.resolve_pkg_reqs(&all_package_reqs).await;
+        match resolutions {
+          NpmPackageReqsResolution::Resolutions(resolutions) => {
+            assert_eq!(all_package_reqs.len(), resolutions.len());
+            for (req, resolution) in
+              all_package_reqs.into_iter().zip(resolutions.into_iter())
+            {
+              let items = items_by_req.get(&req).unwrap();
+              for item in items {
+                match &resolution {
+                  Ok(pkg_nv) => {
+                    self
+                      .pending_info
+                      .specifier_resolutions
+                      .insert(item.specifier.clone(), pkg_nv.clone());
+                    let pkg_id_ref =
+                      NpmPackageNvReference::new(PackageNvReference {
+                        nv: pkg_nv.clone(),
+                        sub_path: item
+                          .package_ref
+                          .sub_path()
+                          .map(ToOwned::to_owned),
+                      });
+                    let resolved_specifier = pkg_id_ref.as_specifier();
+                    if resolved_specifier != item.specifier {
+                      self.pending_info.redirects.insert(
+                        item.specifier.clone(),
+                        resolved_specifier.clone(),
+                      );
+                    }
+                    self.pending_info.module_slots.insert(
+                      resolved_specifier.clone(),
+                      ModuleSlot::Module(Module::Npm(NpmModule {
+                        specifier: resolved_specifier,
+                        nv_reference: pkg_id_ref,
+                      })),
+                    );
+                  }
+                  Err(err) => {
+                    self.pending_info.module_slots.insert(
+                      item.specifier.clone(),
+                      ModuleSlot::Err(ModuleError::LoadingErr(
+                        item.specifier.clone(),
+                        item.maybe_range.clone(),
+                        NpmLoadError::PackageReqResolution(err.clone()).into(),
+                      )),
+                    );
+                  }
+                }
+              }
+            }
+          }
+          NpmPackageReqsResolution::ReloadRegistryInfo => {
+            restart_count += 1;
+
+            if restart_count > 1 {
+              // makes no sense for someone to reload registry information more than once
+              panic!("cannot return back ReloadRegistryInfo more than once");
```

Comment:
Should we bring the process down? Maybe return an error so that CLI can handle it more cleanly

Reply:
I updated it to not panic.